salt/cloud/libcloudfuncs.py

```diff
@@ -274,17 +276,18 @@ def get_image(conn, vm_):
     '''
     images = conn.list_images()
 
-    vm_image = config.get_cloud_config_value('image', vm_, __opts__).encode(
-        'ascii', 'salt-cloud-force-ascii'
-    )
+    if not six.PY3:
+        vm_image = config.get_cloud_config_value('image', vm_, __opts__).encode(
```

Comment:
Don't we still need a `vm_image` set if we are running python 3? Similar to the change in line 246 above: https://github.com/saltstack/salt/pull/34105/files#diff-e8c33e1c4a43de8e11379a6d5cf064b0R246